### PR TITLE
Add bvec reorientation functionality to Utilities CLI

### PR DESCRIPTION
**Pull request by AlexanderGetka-cbica**
Addresses #840 

A new option is added to the CLI for Utilities.exe, used with the parameter -bv (or --bvec).
When passed alongside the other required parameters for reorienting an image with the -or option, it will accept the filename of a corresponding bvec file which will also be changed to the user-specified desired output orientation. The bvec output of this reorientation will be written to a file with the same base name as the reoriented output image, but with the .bvec file extension.

This functionality is only available for 4D images based on my understanding of the use case. If this is needed for other image types please let me know.

This also adds some slight clarifications to the usage info and command-line output.

### CODE REVIEW

File: src/applications/Utilities/src/Utilities.cxx [hunk 1]
```diff
@@ -1437,6 +1515,26 @@ int main(int argc, char** argv)
...
         cbica::WriteImage< ImageType >(joinFilter->GetOutput(), outputImageFile);
       }
+      std::cout << "Finished reorienting " + inputImageFile + " (" + originalOrientation + ") to "
+          + outputImageFile + " (" + orientationDesired + ")" << std::endl;
+      if (!inputBvecFile.empty())
+      {
+          std::cout << "Reorienting supplied bvec file to " + orientationDesired << std::endl;
+          std::string bvecOutputFile = path + "/" + base + ".bvec"; // same basename as output image
+
+          double transform[9] = { 0.0 };
```
Comment (ashishsingh18) [inline src/applications/Utilities/src/Utilities.cxx:1525]: add a brief comment that the code is replicating the steps in python script

File: src/applications/Utilities/src/Utilities.cxx [hunk 2]
```diff
@@ -1437,6 +1515,26 @@ int main(int argc, char** argv)
...
+          MatrixType resultMatrix(dataMatrix); // same shape as data
+          resultMatrix = dataMatrix.transpose() * transformMatrix; // apply the transform
+          resultMatrix.inplace_transpose(); // transpose back to original shape
+          bool writeSucceeded = WriteBvecFile(resultMatrix, bvecOutputFile);
+          if (!writeSucceeded) 
+          {
+              return EXIT_FAILURE;
+          }
+      }
       return EXIT_SUCCESS;
```
Comment (ashishsingh18) [inline src/applications/Utilities/src/Utilities.cxx:1538]: add a cout msg saying that the bvec reorientation is done.

File: src/applications/Utilities/src/Utilities.cxx [hunk 3]
```diff
@@ -856,7 +910,10 @@ int main(int argc, char** argv)
...
   parser.addOptionalParameter("dsJ", "dcmSegJSON", cbica::Parameter::FILE, "JSON file for Metadata", "The extra metadata needed to generate the DICOM-Seg object", "Use http://qiicr.org/dcmqi/#/seg to create it", "Use '-i' to pass input NIfTI segmentation image", "Use '-o' to pass output DICOM file");
-  parser.addOptionalParameter("or", "orient", cbica::Parameter::STRING, "Desired 3 letter orientation", "The desired orientation of the image", "See the following for supported orientations (use last 3 letters only):", "https://itk.org/Doxygen/html/namespaceitk_1_1SpatialOrientation.html#a8240a59ae2e7cae9e3bad5a52ea3496e");
+  parser.addOptionalParameter("or", "orient", cbica::Parameter::STRING, "Desired 3 letter orientation", "The desired orientation of the image in \"from\" convention", "See the following for supported orientations (use last 3 letters only):", "https://itk.org/Doxygen/html/namespaceitk_1_1SpatialOrientation.html#a8240a59ae2e7cae9e3bad5a52ea3496e",
```
Comment (ashishsingh18) [inline src/applications/Utilities/src/Utilities.cxx:913]: user would not know what the 'from convention' is.

**Comment by AlexanderGetka-cbica** [comment]
Hey guys. I made some minor edits to make it clearer where we got the bvec reorientation concept from. This includes a "deprecated" directory which is meant to house old implementations in python, matlab, etc. I think this should suffice, but let me know if you have ideas to make this better. 

(Same with the usage text edits -- if you can think of an easier way to word the convention-related stuff, let's tweak it!)

**Comment by ashishsingh18** [comment]
@PhucNgo1711 Any ideas why the windows build on Azure is stuck? It has happened on my PR as well.

**Comment by ashishsingh18** [comment]
@AlexanderGetka-cbica This PR needs to be re-triggered.

**Comment by ashishsingh18** [comment]
@CBICA/devteam Can you review this PR?

**Pull request merged**